### ISSUE TITLE
Linking to documentation heading

### ISSUE THREAD

**Issue by LecrisUT** [executablebooks/MyST-Parser#847]
### Describe the gap in the current documentation

Let's say that we want to link to [`sphinx:usage/quickstart#intersphinx`](https://www.sphinx-doc.org/en/master/usage/quickstart.html#intersphinx). Navigating through sphinx's `objects.inv` we find the relative object `std.doc.usage/quickstart` pointing to the page we want to reference, but then, how do we go from there to use the documentation page heading? The closest thing we find in the [documentation](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#reference-roles) is `std.label.code-examples`, but in this case it does not have an explicit label to reference. 

Basically, what would the user use in this situation?
```markdown
[Link to doc][usage/quickstart]
[Link to doc#heading][intersphinx]

[usage/quickstart]: inv:sphinx:std:doc#usage/quickstart
<!-- [intersphinx]: inv:sphinx:std:doc#usage/quickstart#interpshinx -->
[intersphinx]: https://www.sphinx-doc.org/en/master/usage/quickstart.html#intersphinx
```
